source/io/module_javaScriptInvoke.js

```diff
@@ -237,21 +235,44 @@ var assignJavaScriptInvoke;
 
         var returnValueVisitor = createJavaScriptInvokeReturnValueVisitor();
 
-        var findJavaScriptFunctionToCall = function (functionName, isInternalFunction) {
-            if (isInternalFunction) {
-                return {
-                    functionToCall: internalFunctionsMap.get(functionName),
-                    context: undefined
-                };
-            }
-
+        var getGlobal = function () {
             // Normally we do not use typeof checks to see if a value is undefined
             // however in this case it is used to prevent ReferenceErrors when probing global objects
-            var jsSelfScope = typeof self !== 'undefined' ? self : {};
-            var jsGlobalScope = typeof global !== 'undefined' ? global : jsSelfScope;
-            var context = typeof window !== 'undefined' ? window : jsGlobalScope;
+            if (typeof window !== 'undefined') {
+                return window;
+            } else if (typeof self !== 'undefined') {
+                return self;
+            } else if (typeof global !== 'undefined') {
+                return global;
+            }
+            return {};
+        };
+        var internalFunctionsMap = new Map();
+        var jsInvokeGlobal = getGlobal();
```

Comment:
Is there a reason why we don't use the term `scope` anymore? The new names that just end in `global` and not `globalScope` are slightly confusing, given the string in the JSLI document is also called a "global"

Reply:
Talked with sid and agreed that the name global is okay. The global object is a an object you can get a reference to and the properties of it are made accessible in all scopes. You cannot hold a reference to "the global scope" but you can hold a reference to the global object.